irohad/ametsuchi/temporary_wsv.hpp

```diff
@@ -51,7 +51,7 @@ namespace iroha {
        */
       virtual bool apply(const model::Transaction &transaction,
                          std::function<bool(const model::Transaction &,
-                                            CommandExecutor &, WsvQuery &)>
+                                            WsvCommand &, WsvQuery &)>
                              function) = 0;
```

Comment:
Yes, each concrete command will implement `validate(WsvQuery, Context)` and `execute(WsvQuery, WsvCommand, Context)`, methods.
In apply, you will call something like:
```
for (auto cmd : tx.commands){
  // define command Context
  if (cmd.validate(WsvQuery, Context)){
    cmd.execute(WsvQuery, WsvCommand, Context);
  }
}
```